### ISSUE TITLE
`id` should be passed to `FormControl` in the `Field` component

### ISSUE THREAD

**Issue by kculmback-eig** [saas-js/saas-ui#33]
Right now if you pass an `id` prop to the `Field` component, the `id` is passed only to the `Input`, `Select`, etc component. Chakra is unaware of those so it generates its own `id` and `htmlFor` props for the `FormLabel` component. To ensure fields are fully accessible, the `id` prop should instead be passed to `FormControl` component so that Chakra uses the proper `id` and `htmlFor` attributes for the label, input, etc.

Example code below. I was unable to test this because I could not get the `saas-ui` project running locally (yarn install failed because I currently have not purchased saas ui pro).

```typescript
// packages/saas-ui-forms/src/field.tsx

export const BaseField: React.FC<FieldProps> = (props) => {
  const { id, name, label, help, variant, hideLabel, children, ...controlProps } =
    props

  // ...

  return (
    <FormControl id={id} variant={variant} {...controlProps} isInvalid={!!error}>
      // ...
    </FormControl>
  )
}

const createField = (
  InputComponent: React.FC<any>,
  { displayName, hideLabel, BaseField }: CreateFieldProps
) => {
  const Field = forwardRef<FieldProps, typeof FormControl>((props, ref) => {
    const {
      id,
     // ...
    } = props

    return (
      <BaseField
        id={id}
        // ...
      >
        // ...
      </BaseField>
    )
  })

  // ...

  return Field
}

**Comment by Pagebakers** [comment]
Thanks for reporting @kculmback-eig ! 

I had this issue with the PasswordInput component. This got fixed, but hadn't experienced this with the other other fields.

I will look into this.

PS,

You can remove @saas-ui/pro from apps/website/package.json to make it work. It's added to optionalDependencies but somehow yarn fails to install. Still looking for a solution for this.

**Comment by Pagebakers** [comment]
@kculmback-eig Fixed in 0.13.3

Thanks again!

**Comment by kculmback-eig** [comment]
Awesome!
